src/devSVG.cpp

```diff
@@ -429,10 +495,25 @@ void svg_path(double *x, double *y,
 double svg_strwidth(const char *str, const pGEcontext gc, pDevDesc dd) {
   SVGDesc *svgd = (SVGDesc*) dd->deviceSpecific;
 
-  gdtools::context_set_font(svgd->cc, fontname(gc->fontfamily, gc->fontface),
-    gc->cex * gc->ps, is_bold(gc->fontface), is_italic(gc->fontface));
+  std::string file = fontfile(gc->fontfamily, gc->fontface, svgd->font_aliases, svgd->font_extra);
+  gdtools::context_set_font(svgd->cc, fontname(gc->fontfamily, gc->fontface, svgd->font_aliases),
+                            gc->cex * gc->ps, is_bold(gc->fontface), is_italic(gc->fontface), file);
   FontMetric fm = gdtools::context_extents(svgd->cc, std::string(str));
 
+  if (fm.width == 0) {
+    int length = 0;
+    if (mbcslocale) {
+      // Compute width of UTF8 string
+      const char* s = str;
+      while (*s)
```

Comment:
Could you explain the logic here?

Alternatively, it looks R uses `int nc = (int) mbstowcs(NULL, translateChar(string), 0);` in `R_nchar()` so maybe we could also use that?